app/uk/gov/hmrc/exports/services/reversemapping/declaration/transport/TransportParser.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services.reversemapping.declaration.transport
+
+import uk.gov.hmrc.exports.models.declaration.Transport
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser
+import uk.gov.hmrc.exports.services.reversemapping.declaration.DeclarationXmlParser.XmlParserResult
+
+import javax.inject.Inject
+import scala.xml.NodeSeq
+
+class TransportParser @Inject()(containersParser: ContainersParser) extends DeclarationXmlParser[Transport] {
+
+  override def parse(inputXml: NodeSeq): XmlParserResult[Transport] =
+    for {
```

Comment:
Isn't the `for` loop the same as just
```
    containersParser
      .parse(inputXml)
      .map(containers => Transport(containers = Option(containers)))
```
It seems a bit over-engineered otherwise

Reply:
There are going to be many other parsers being called here in future (like with the `ExportsDeclarationXmlParser`) so following the same pattern to support multiple parser calls to populate the Transport model object

Reply:
Ok the `for` loop , but still, why the pattern matching?
Why not just...
```
for {
  containers <- containersParser.parse(inputXml)
} yield Transport(containers = Option(containers))
```